Added atcoder support

sainad2222 opened the pull request:
added atcoder to api
json fields:
rating
highest(highest rating)
rank
level

sainad2222 reviewed (commented):
Everything's fine when tested in local environment

Abhijeet-AR commented on details_soup.py:
```diff
@@ -296,6 +298,36 @@ def __interviewbit(self):
+            raise UsernameError("User not Found")
+
+        soup = BeautifulSoup(page.text, "html.parser")
+        table = soup.find_all("table", class_="dl-table")[1]
```
Handle the case where the no.of tables are less than 2.

Abhijeet-AR replied:
Sorry for not being clear. The reason I wanted to handle this was to not throw an error for a new user at AtCoder as there is no table for a user without any rank, rating, ...

Abhijeet-AR replied:
You can check for the table size and handle it before try, except.

sainad2222 replied:
What should I return if it's new user? How about a json response with details of just "new user"

Abhijeet-AR commented on details_soup.py:
```diff
@@ -296,6 +298,36 @@ def __interviewbit(self):
+        details = {
+            "status": "Success",
+            "username": self.__username,
+            "platform": "Atcoder",
```
We don't need platform in our response. Let's follow the same format for all the APIs.

Abhijeet-AR commented:
> Everything's fine when tested in local environment

Thanks for the PR @sainad2222. I found a couple of issues, let's fix it and merge.

sainad2222 commented:
Regarding BrokenChangesError if there are some broken changes like change in UI or selector during site updates that may break code. I now added code to handle this exception and also added table issue in try block. Let me know if any changes needed. Thanks for API :)

saurabh-prakash reviewed (approved):
:shipit:

Abhijeet-AR commented:
You forgot to import `BrokenChangesError` in `main.py`.

Abhijeet-AR commented on details_soup.py:
```diff
@@ -307,9 +307,9 @@ def __atcoder(self):
-        table = soup.find_all("table", class_="dl-table")[1]
-        rows = table.find_all("td")
         try:
+            table = soup.find_all("table", class_="dl-table")[1]
```
Sorry for not being clear. The reason I wanted to handle this was to not throw an error for a new user at AtCoder as there is no table for a user without any rank, rating, ...

Abhijeet-AR replied:
You can check for the table size and handle it before try, except.

sainad2222 replied:
What should I return if it's new user? How about a json response with details of just "new user"

sainad2222 replied:
@Abhijeet-AR ?

Abhijeet-AR replied:
Yeah. You can just return the details and in place of rank, and other stuff you can return a NA

Abhijeet-AR commented:
> Regarding BrokenChangesError if there are some broken changes like change in UI or selector during site updates that may break code. I now added code to handle this exception and also added table issue in try block. Let me know if any changes needed. Thanks for API :)
> 
> If you feel like the name BrokenChangesError is not relevant feel free to suggest new name. I'll change accordingly. I feel like it's necessary to handle these type of exceptions too

Yeah, that's a good idea. Completely makes sense.

sainad2222 commented:
I tested once again locally. Looks fine to me. Let me know if any changes needed
